### DISCUSSION TITLE
long urls start failing above 4k characters

### DISCUSSION THREAD

**Discussion by hst-m** [uNetworking/uWebSockets.js#314] [Q&A] [answered]
I've got some long query strings on some urls and once they start getting above around 4k characters the uWS response fails with `(failed) net::ERR_EMPTY_RESPONSE` and the url callback does not get called, and there are no errors displayed in the server logs

```js
uws.App().any('/*', (res,req) => console.log('request'))
```
simplified example where `request` does not get logged and no errors logged with request urls above around 4k length, the issue happens maybe 50% of the time at this length and when you get up around 15k it happens about 100% of the time. Are there any limits on url length that I am missing in the docs somewhere? Technically the url spec has no limit on url length. let me know if need any more info

I will likely add solution to reduce url length but wondering if this is issue that needs to be fixed

**Comment by ghost** [comment, marked as answer]
We use a 4kb max length for requests, meaning URLs must be less than 4kb. This is for security reasons.

> Technically the url spec has no limit on url length.

It also is not guaranteed to be ANY particular length, it's entirely up to the implementation.

**Comment by alexandrebini** [comment]
Is the 4kb max length still valid? I'm able to make it work but I'm getting intermittent empty responses (and the request never reaches the handler).

If that limit is still true, is there a way to tweak it?

Thanks!!
